src/main/java/com/stripe/net/FormEncoder.java

```diff
@@ -18,6 +16,30 @@
 import java.util.stream.Collectors;
 
 final class FormEncoder {
+  public static HttpContent createHttpContent(Map<String, Object> params) throws IOException {
+    // If params is null, we create an empty HttpContent because we still want to send the
+    // Content-Type header.
+    if (params == null) {
+      return HttpContent.buildFormURLEncodedContent(new ArrayList<KeyValuePair<String, String>>());
+    }
+
+    Collection<KeyValuePair<String, Object>> flatParams = flattenParamsValue(params, null);
+
+    // If all parameters have been encoded as strings, then the content can be represented
+    // with application/x-www-form-url-encoded encoding. Otherwise, use
+    // multipart/form-data encoding.
+    if (flatParams.stream().allMatch(kvp -> kvp.getValue() instanceof String)) {
+      Collection<KeyValuePair<String, String>> flatParamsString =
+          flatParams.stream()
+              .filter(kvp -> kvp.getValue() instanceof String)
+              .map(kvp -> new KeyValuePair<String, String>(kvp.getKey(), (String) kvp.getValue()))
+              .collect(Collectors.toList());
```

Comment:
Man, I'm jealous of your Very Modern Java skills (skillz?) ;)

Reply:
😂